zrml/court/src/types.rs

```diff
@@ -0,0 +1,375 @@
+// Copyright 2021-2022 Zeitgeist PM LLC.
+//
+// This file is part of Zeitgeist.
+//
+// Zeitgeist is free software: you can redistribute it and/or modify it
+// under the terms of the GNU General Public License as published by the
+// Free Software Foundation, either version 3 of the License, or (at
+// your option) any later version.
+//
+// Zeitgeist is distributed in the hope that it will be useful, but
+// WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+// General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Zeitgeist. If not, see <https://www.gnu.org/licenses/>.
+extern crate alloc;
+use alloc::{vec, vec::Vec};
+use zeitgeist_primitives::types::OutcomeReport;
+
+/// The type of the court identifier.
+pub type CourtId = u128;
+
+/// The different court vote types. This can be extended to allow different decision making options.
+#[derive(
+    parity_scale_codec::Decode,
+    parity_scale_codec::Encode,
+    parity_scale_codec::MaxEncodedLen,
+    scale_info::TypeInfo,
+    Clone,
+    Debug,
+    PartialEq,
+    Eq,
+)]
+pub enum VoteItemType {
+    Outcome,
+    Binary,
+}
+
+/// The different court vote types with their raw values.
+/// This can be extended to allow different decision making options.
+#[derive(
+    parity_scale_codec::Decode,
+    parity_scale_codec::Encode,
+    parity_scale_codec::MaxEncodedLen,
+    scale_info::TypeInfo,
+    Clone,
+    Debug,
+    PartialEq,
+    Eq,
+    Ord,
+    PartialOrd,
+)]
+pub enum VoteItem {
+    Outcome(OutcomeReport),
+    Binary(bool),
+}
```

Comment:
> I would have expected market-commons to have completely disappeared from this pallet.

I would have expected that too, but in practice I started with using market ids instead of general court ids.

Reply:
I don't know. I'll let this review comment open until I fixed the other ones for now.

Reply:
How are scalar markets handled with that?

Reply:
https://github.com/zeitgeistpm/zeitgeist/issues/1004